tests/data/sample_specification.yaml

```diff
@@ -8,36 +8,37 @@ aea_version: '>=1.0.0, <2.0.0'
 description: 'A protocol for testing purposes.'
 speech_acts:
   performative_ct:
-    content_ct: ct:DataModel
+    content_ct: ct:DataModel7
   performative_pt:
     content_bytes: pt:bytes
     content_int: pt:int
     content_float: pt:float
     content_bool: pt:bool
     content_str: pt:str
   performative_pct:
-#    content_set_ct: pt:set[ct:DataModel] # custom type inside of set, list, and dict isn't allowed.
     content_set_bytes: pt:set[pt:bytes]
     content_set_int: pt:set[pt:int]
     content_set_float: pt:set[pt:float]
     content_set_bool: pt:set[pt:bool]
     content_set_str: pt:set[pt:str]
-#    content_list_ct: pt:list[ct:DataModel] # custom type inside of set, list, and dict isn't allowed.
+    # content_set_ct: pt:set[ct:DataModel] # custom type inside of set isn't allowed.
+    content_list_ct: pt:list[ct:DataModel6]
+    content_list_ct2: pt:list[ct:DataModel3]
     content_list_bytes: pt:list[pt:bytes]
     content_list_int: pt:list[pt:int]
     content_list_float: pt:list[pt:float]
     content_list_bool: pt:list[pt:bool]
     content_list_str: pt:list[pt:str]
   performative_pmt:
-#    custom type inside of set, list, and dict isn't allowed.
-#    content_dict_int_ct: pt:dict[pt:int, ct:DataModel]
-#    content_dict_ct_ct: pt:dict[ct:DataModel, ct:DataModel]
 #    invalid in protobuf (key in map<X, Y> cannot be 'bytes', 'float', 'double', 'message')
 #    content_dict_bytes_bytes: pt:dict[pt:bytes, pt:bytes]
 #    content_dict_bytes_int: pt:dict[pt:bytes, pt:int]
 #    content_dict_bytes_float: pt:dict[pt:bytes, pt:float]
 #    content_dict_bytes_bool: pt:dict[pt:bytes, pt:bool]
 #    content_dict_bytes_str: pt:dict[pt:bytes, pt:str]
+#    content_dict_ct_ct: pt:dict[ct:DataModel, ct:DataModel]  # not allowed
+    content_dict_int_ct: pt:dict[pt:int, ct:DataModel]
+    content_dict_int_ct2: pt:dict[pt:int, ct:DataModel2]
```

Comment:
This is just like the line above it. Only 1 of these is enough.

Reply:
Add versions with other keys and a custom type value.

Reply:
done